optimum/habana/transformers/trainer.py

```diff
@@ -142,6 +142,13 @@ def __init__(
             logger.info(f"No `GaudiTrainingArguments` passed, using `output_dir={output_dir}`.")
             args = GaudiTrainingArguments(output_dir=output_dir)
 
+        from dataclasses import asdict
+        if args.half_precision_backend is not None and args.half_precision_backend == 'hpu_amp':
+            delattr(args, 'half_precision_backend')
+            self.use_hpu_amp = True
+        else:
+            self.use_hpu_amp = False
```

Comment:
Autocast enablement is supposed to be an HMP substitution not an alternative. I' would like to decouple those two approaches as it is easier to remove any HMP related code in case of feature deprecation. This way we also stick closer to
 🤗 Transfromers. I believe we should use GaudiConfig as little as possible if there is a way to reflect native Transformers usage. I stand by my change because of those reasons, but of course if you think your approach is more practical/sensible I will change it :)

Reply:
Interesting, I had the feeling HMP would be there longer :sweat_smile: 
I have a sync with Sree next Tuesday, I'll let you know what we decide to do moving forward :+1: 
In the meantime, I'll try to run some regression tests on this PR.

Reply:
So we decided to keep the Transformers way that you proposed :slightly_smiling_face: 
I'm going to resolve comments that are linked to the Gaudi config and review this PR again.